pipeline/metadata/ip_metadata_chooser.py

```diff
@@ -0,0 +1,51 @@
+"""A module which contains a number of sources of ip metadata and chooses how to add them to ips."""
+import datetime
+import logging
+
+from pipeline.metadata.flatten import Row
+from pipeline.ip_metadata_values import IpMetadataValuesInterface
+
+
+class IpMetadataChooser():
+  """Business logic for selecting IP metadata values from a number of sources"""
+
+  def __init__(self, date: datetime.date,
+               prod_values: IpMetadataValuesInterface) -> None:
```

Comment:
Pass the dbip, maxmind and caida objects here instead. No need to pass the folders.

Reply:
I can't pass the objects because they can't be initialized yet at this point.

I've switched to passing in the classes/folders and having a lazy init that takes the date.